src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -9201,6 +9206,15 @@ def quantiles_along_axis0(
             ErrorMessage.not_implemented(
                 "quantile is not supported for datetime columns"
             )
+        assert index is None or len(index) == len(
+            q
+        ), f"length of index {index} did not match quantiles {q}"
+        if len(query_compiler.columns) == 1:
+            # Use method without UNION ALL operations if the query compiler has only a single column
+            index_values = q if index is None else index
```

Comment:
can we add a comment to explain the if expression here?

Reply:
added (also moved further up)